pkg/executables/cmk.go

```diff
@@ -119,39 +139,136 @@ func (c *Cmk) ValidateAffinityGroupsPresent(ctx context.Context, domain string,
 	return nil
 }
 
-func (c *Cmk) ValidateZonePresent(ctx context.Context, zone v1alpha1.CloudStackResourceRef) error {
+func (c *Cmk) ValidateZonesPresent(ctx context.Context, zones []v1alpha1.Zone) error {
```

Comment:
This signature seems off. Why does the `Cmk` executable care about multiple zones? I would think the fact there are multiple zones that need validating is a property of some validation logic and this signature is just for 1 zone reducing its scope.

```go
func (c *Cmk) ZoneExists(ctx context.Context, zone v1alpha1.Zone) error
```

Reply:
OK, I've just noticed there are a bunch of existing `Validate*` methods on this struct. Whats the intent behind this struct because it contains both rules on whats OK and not OK as well as information retrieval from CloudStack so conflates several concerns.

@g-gaston is this on your radar?

Reply:
@chrisdoherty4 I agree with your concerns. 
CMK should only provider some functions to call cloudstack to retrieve some info back and should not do extra, like validation. 
The validation should be inside validator.go file. 

I am thinking do the refactor in next PR since Max and I are both touching this cmk file in on-going PRs.